File: backend/src/models/Tweet.ts
import {LightWeightUser} from "./LightWeightUser";

// Global secondary index - `user_id`
export interface Tweet {
    id: string,
    userId: string,
    user: LightWeightUser
    tweet: string,
    createdAt: string,
    likeList: Array<LightWeightUser>,
    imageUrl?: string
}

File: android/app/src/main/java/app/syam/twitter/tweet/model/Tweet.kt
package app.syam.twitter.tweet.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Tweet(
    @SerializedName("user")
    val user: LightWeightUser?,
    @SerializedName("createdAt")
    val createdAt: String?,
    @SerializedName("id")
    val id: String?,
    @SerializedName("likeList")
    val likeList: List<LightWeightUser>?,
    @SerializedName("userId")
    val userId: String?,
    @SerializedName("tweet")
    val tweet: String?,
    @SerializedName("imageUrl")
    val imageUrl: String?
): Parcelable